### PR TITLE
Integration test for XACML based scope validator for access token.

### PR DESCRIPTION
**Pull request by senthalan**

### CODE REVIEW

**Comment by nilasini** [comment]
Please resolve the conflicts

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+        String username = "admin";
+        Secret password = new Secret("admin");
+        AuthorizationGrant passwordGrant = new ResourceOwnerPasswordCredentialsGrant(username, password);
+
+        ClientID clientID = new ClientID(consumerKey);
+        Secret clientSecret = new Secret(consumerSecret);
+        ClientAuthentication clientAuth = new ClientSecretBasic(clientID, clientSecret);
+
+
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java]: Do we need this new line

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+     * Request access token with the scope and validate the token.
+     *
+     * @param scope scope
+     * @return whether validation success or not
+     * @throws Exception exception
+     */
+    private boolean getTokenAndValidate(Scope scope) throws Exception {
+
+        // Reset client.
+        client = HttpClientBuilder.create().disableRedirectHandling().build();
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java]: Don't we need to close the client in a finally block?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.identity.application.common.model.xsd.ServiceProvider;
+import org.wso2.carbon.identity.entitlement.stub.dto.PolicyDTO;
+import org.wso2.carbon.identity.oauth.stub.dto.OAuthConsumerAppDTO;
+import org.wso2.identity.integration.common.clients.entitlement.EntitlementPolicyServiceClient;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.net.URI;
+
+public class OAuth2XACMLScopeValidatorTestCase extends OAuth2ServiceAbstractIntegrationTest {
```
Comment (mefarazath) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java]: Add a class level comment please explaining what you expect to test with this test class

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        OAuthConsumerAppDTO oAuthConsumerAppDTO = getBasicOAuthApp(CALLBACK_URL);
+        oAuthConsumerAppDTO.setScopeValidators(new String[]{XACML_BASED_SCOPE_VALIDATOR_NAME});
+        ServiceProvider serviceProvider = registerServiceProviderWithOAuthInboundConfigs(oAuthConsumerAppDTO);
+        Assert.assertNotNull(serviceProvider, "OAuth App creation failed.");
+        Assert.assertNotNull(consumerKey, "Consumer Key is null.");
+        Assert.assertNotNull(consumerSecret, "Consumer Secret is null.");
+    }
+
+    @Test(groups = "wso2.is", description = "Check publishing a policy", dependsOnMethods = "testRegisterApplication")
+    public void testPublishPolicy() throws Exception {
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java]: Please add a new line

Comment (senthalan) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java]: done

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+import org.wso2.identity.integration.common.clients.entitlement.EntitlementPolicyServiceClient;
+import org.wso2.identity.integration.test.utils.OAuth2Constant;
+
+import java.net.URI;
+
+/**
+ *  Test cases to check the functionality of the XACML based scope validator.
+ *  Register OAuth app with XACML based scope validator enabled then get access tokens with various scopes and validate.
+ */
+public class OAuth2XACMLScopeValidatorTestCase extends OAuth2ServiceAbstractIntegrationTest {
```
Comment (nilasini) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2XACMLScopeValidatorTestCase.java]: Please add a new line

File: pom.xml [hunk 6]
```diff
@@ -1702,7 +1702,7 @@
         <identity.notification.json.version>5.1.1</identity.notification.json.version>
 
         <!-- Forget-me tool -->
-        <forgetme.tool.version>1.1.8</forgetme.tool.version>
+        <forgetme.tool.version>1.1.6</forgetme.tool.version>
```
Comment (nilasini) [inline pom.xml]: Is there any specific reason to reduce the version?

**Pull request merged**